solidjs-tailwind/src/components/FileExplorer/FileExplorer.jsx

```diff
@@ -0,0 +1,52 @@
+import { For } from 'solid-js';
+import { Link } from '@solidjs/router';
+import { removeLastPathPart } from '../../utils/removeLastPathPart';
+import { document } from 'solid-heroicons/outline';
+import { folder } from 'solid-heroicons/solid';
+import { Icon } from 'solid-heroicons';
+import styles from './FileExplorer.module.css';
+
+const FileExplorerView = (props) => {
+  return (
+    <div class={styles.container}>
+      {props.repoPath && (
+        <Link
+          href={`${props.basePath}/tree/${props.branch}/${removeLastPathPart(
+            props.repoPath
+          )}`}
+        >
+          <a class={styles.cellBack}>
+            <div class="text-blue-600">..</div>
+          </a>
+        </Link>
+      )}
+      <For each={props.items}>
+        {(item) => (
+          <div class={styles.cell}>
+            <div class="flex items-center">
+              <div class="mr-2.5">
+                {item.type === 'tree' ? (
+                  <Icon path={folder} class={styles.iconDir} />
+                ) : (
+                  <Icon path={document} class={styles.iconFile} />
+                )}
+              </div>
+              <Link
+                href={`${props.basePath}/${item.type}/${props.branch}/${item.path}`}
```

Comment:
lets make this a variable too

Reply:
This will mean creating a function and passing the mapped item into it. Just like the other one, I don't think it is needed.